app/helpers/datasets_helper.rb

```diff
@@ -10,7 +10,7 @@ module DatasetsHelper
 
   def edit_dataset_url(dataset)
     url = URI::HTTPS.build(host: 'data.gov.uk')
-    url += if dataset.datafiles.none?
+    url += if !dataset.released?
```

Comment:
How about you flip the conditions to avoid the negation?

```rb
url += if dataset.released?
  '/dataset/edit/'
else
  '/unpublished/edit-item/'
end
```